### DISCUSSION TITLE
2.10.4, Slider steps on single left click from touchpad or touchscreen do not match settings. Windows 11.

### DISCUSSION THREAD

**Discussion by oldsparks** [artisan-roaster-scope/artisan#1610] [General]
I've found that despite a slider step set to 5 or 1, when the slider is clicked once above or below the horizontal bar, the value jumps by 10. The slider steps work as defined if it is selected and dragged up or down. So a slider set to single steps with a zero to 100 range can be set to 62 for instance, but then if a single left click above or below the bar the slider will jump to 72 or 52. 

I am using either a touchpad or the touchscreen. I have not tried with a mouse as I no longer even own a mouse.

**Comment by poundy** [comment]
I'm pretty sure you're mis-interpreting what that setting is. That setting is the increments possible on a slider - so a value of 5 in there would mean that the value could only go up in 5's, so it would never move from 25 to 28, it'd go to 30.  What you're doing through your action is making it scroll. That's a separate thing, and it's expected to move it by larger increments to allow you to rapidly change (scroll) values. I suspect there is nothing you can do to change that behaviour, it's more an OS thing than an Artisan thing...

**Reply by oldsparks** [reply]
I understand the setting. I guess I just don't understand the difference of "scroll" in Windows. Nor do I fully understand why it "scrolls" in increments of 10. I don't know if it's programmatically possible to have the steps setting extend to that feature of windows. It would certainly be more useful to be able to just rapidly click above or below the slider bar and move the slider by the set step value vice increments of 10. Oh well, just an observation that left me perplexed. Thanks.

**Comment by MAKOMO** [comment]
The behavior seems to be very OS dependent. Internally those sliders have two configuration, singleStep and pageStep. The singleStep is set by the Artisan slider configuration to 1, 5 or 10. The pageStep is automatically set to 10 if singleStep is 1 or 5 and 20 otherwise. The pageStep is applied if using the page up/down keys, the singleStep is applied if using the cursor up/down keys. A click to the sliders bar on macOS moves the slider to the closest singleStep position (so if singleStep is 5 this is 55 if clicked on position 53). This behavior seems to be different on Windows. Nothing we can do about.

**Comment by oldsparks** [comment]
Ah, interesting difference between the OS's there. I appreciate the background on this.

-David
